docs/guides/ci/gh-actions.md

```diff
@@ -0,0 +1,216 @@
+
+If your test coverage isn't great, consider at least adding an example program, and building it as part of CI:
+
+For a library:
+
+```yaml
+          - name: Build example
+            run: crystal build examples/hello.cr
+```
+
+For an application (very good to do even if you have specs):
+
+```yaml
+          - name: Build
+            run: crystal build src/game_of_life.cr
+```
+
+### Testing with different vesions of Crystal
+
+By default, the latest released version of Crystal is installed. But you may want to also test with the "nightly" build of Crystal, and perhaps some older versions that you still support for your project. Change the top of the workflow as follows:
+
+```{.yaml hl_lines="6 14"}
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        crystal: [0.35.1, latest, nightly]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download source
+        uses: actions/checkout@v2
+      - name: Install Crystal
+        uses: oprypin/install-crystal@v1
+        with:
+          crystal: ${{ matrix.crystal }}
+      - ...
+```
+
+All those versions will be tested for *in parallel*.
+
+By specifying the version of Crystal you could even opt *out* of supporting the latest version (which *is* a moving target), and only support particular ones.
+
+### Testing on multiple operating systems
+
+Typically, developers run tests only on Ubuntu, which is OK if there is no platform-sensitive code. But it's easy to add another [system](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources) into the test matrix, just add the following near the top of your job definition:
+
+```{.yaml hl_lines="6 7"}
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - ...
+```
+
+## Installing Shards packages
+
+Most projects will have external dependencies, ["shards"](https://github.com/crystal-lang/shards#usage). Having declared them in `shard.yml`, just add the installation step into your workflow (after `install-crystal` and before any testing):
+
+```yaml
+      - name: Install shards
+        run: shards install
+```
+
+### Latest or locked dependencies?
+
+If your repository has a checked in `shard.lock` file (typically good for applications), consider the effect that this has on CI: `shards install` will always install the exact versions specified in that file. But if you're developing a library, you probably want to be the first to find out in case a new version of a dependency breaks the installation of your library -- otherwise the users will, because the lock doesn't apply transitively. So, strongly consider running `shards update` instead of `shards install`. And then it makes sense to add [scheduled runs](https://www.jeffgeerling.com/blog/2020/running-github-actions-workflow-on-schedule-and-other-events) to your repository.
```

Comment:
Perhaps mention that for a library you maybe shouldn't check in `shard.yml` in the first place?

Reply:
OK. I didn't want to express an explicit opinion on that within this guide, but done now.